pennylane/operation.py

```diff
@@ -856,6 +856,10 @@ def _check_batching(self, params):
 
     def __repr__(self):
         """Constructor-call-like representation."""
+        if self.arithmetic_depth > 0:
+            raise NotImplementedError(
+                "Operators with an arithmetic depth greater than zero must define their own __repr__ method"
+            )
```

Comment:
I don't think we need to have this check.  I think we simply just override the method when we need to.